frontend/catalyst/utils/jax_cpu_lapack_kernels/lapack_kernels.cc

```diff
@@ -0,0 +1,914 @@
+{
+    const int32_t lower = *(reinterpret_cast<int32_t *>(data[0]));
+    const int b = *(reinterpret_cast<int32_t *>(data[1]));
+    const int n = *(reinterpret_cast<int32_t *>(data[2]));
+    const T *a_in = reinterpret_cast<T *>(data[3]);
+    const char uplo = lower ? 'L' : 'U';
+
+    void **out = reinterpret_cast<void **>(out_tuple);
+    T *a_out = reinterpret_cast<T *>(out[0]);
+    int *info = reinterpret_cast<int *>(out[1]);
+    if (a_out != a_in) {
+        std::memcpy(a_out, a_in,
+                    static_cast<int64_t>(b) * static_cast<int64_t>(n) * static_cast<int64_t>(n) *
+                        sizeof(T));
+    }
+
+    constexpr int corder = LAPACK_ROW_MAJOR;
+
+    for (int i = 0; i < b; ++i) {
+        *info = fn(corder, uplo, n, a_out, n);
+        a_out += static_cast<int64_t>(n) * static_cast<int64_t>(n);
+        ++info;
+    }
+}
+
+template struct Potrf<float>;
+template struct Potrf<double>;
+template struct Potrf<std::complex<float>>;
+template struct Potrf<std::complex<double>>;
+
+// Gesdd
+
+static char GesddJobz(bool job_opt_compute_uv, bool job_opt_full_matrices)
+{
+    if (!job_opt_compute_uv) {
+        return 'N';
+    }
+    else if (!job_opt_full_matrices) {
+        return 'S';
+    }
+    return 'A';
+}
+
+static int Gesdd_ldu(const int order, const char jobz, const int m, const int n)
+{
+    int ldu = 0;
+    if (jobz == 'N') {
+        ldu = 1;
+    }
+    else if (jobz == 'A') {
+        ldu = m;
+    }
+    else if (jobz == 'S') {
+        if (m >= n) {
+            ldu = (order == LAPACK_ROW_MAJOR) ? n : m;
+        }
+        else {
+            ldu = m;
+        }
+    }
+    return ldu;
+}
+
+static int Gesdd_ldvt(const int order, const char jobz, const int m, const int n)
+{
+    int ldu = 0;
+    if (jobz == 'N') {
+        ldu = 1;
+    }
+    else if (jobz == 'A') {
+        ldu = n;
+    }
+    else if (jobz == 'S') {
+        if (m >= n) {
+            ldu = n;
+        }
+        else {
+            ldu = (order == LAPACK_ROW_MAJOR) ? n : m;
+        }
+    }
+    return ldu;
+}
+
+template <typename T> typename RealGesdd<T>::FnType *RealGesdd<T>::fn = nullptr;
+
+template <typename T> void RealGesdd<T>::Kernel(void *out_tuple, void **data, XlaCustomCallStatus *)
+{
+    const int32_t job_opt_full_matrices = *(reinterpret_cast<int32_t *>(data[0]));
+    const int32_t job_opt_compute_uv = *(reinterpret_cast<int32_t *>(data[1]));
+    const int b = *(reinterpret_cast<int32_t *>(data[2]));
+    const int m = *(reinterpret_cast<int32_t *>(data[3]));
+    const int n = *(reinterpret_cast<int32_t *>(data[4]));
+    const int lwork = *(reinterpret_cast<int32_t *>(data[5]));
+    T *a_in = reinterpret_cast<T *>(data[6]);
+
+    void **out = reinterpret_cast<void **>(out_tuple);
+    T *a_out = reinterpret_cast<T *>(out[0]);
+    T *s = reinterpret_cast<T *>(out[1]);
+    T *u = reinterpret_cast<T *>(out[2]);
```

Comment:
Did you have to switch u and vt?

Reply:
No, when we specify row-order matrix layout and give the correct array leading dimensions `lda`, `ldu` and `ldvt` (following the logic provided in the LAPACK gesdd documentation here: https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-c/2024-2/gesdd.html), then we don't have to swap `u` and `vt` (in fact I'm a little surprised that worked originally, to tell you the truth!)